src/main/java/seedu/jarvis/logic/parser/TaskCommandParser.java

```diff
@@ -63,26 +63,37 @@ public static Deadline parseDeadline(String[] nameKeywords, int length) throws P
         LocalDateTime formattedDeadlineDateTime;
 
         try {
+            String deadlineDescription = parseTimedTaskDescription(nameKeywords, length);
             formattedDeadlineDateTime = parseTimedTaskTime(nameKeywords, length);
+
+            return new Deadline(deadlineDescription, formattedDeadlineDateTime);
+
         } catch (ParseException pe) {
             throw pe;
         }
-
-        String deadlineDescription = parseTimedTaskDescription(nameKeywords, length);
-
-        return new Deadline(deadlineDescription, formattedDeadlineDateTime);
     }
 
     /**
      * Parses timed event description
      * @param nameKeywords
      * @param length
      * @return String of description
+     * @throws ParseException
      */
-    public static String parseTimedTaskDescription(String[] nameKeywords, int length) {
+    public static String parseTimedTaskDescription(String[] nameKeywords, int length) throws ParseException {
         int datePrefixLocation = -1;
+
+        if (nameKeywords[1].substring(0, 2).equals(TASK_DATE)
+            || nameKeywords[1].substring(0, 2).equals(TASK_TIME)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_MISSING_DESCRIPTION));
+        }
+
         for (int i = 2; i < length; i++) {
-            if (nameKeywords[i].substring(0, 2).equals(TASK_DATE)) {
+            if (nameKeywords[i].length() <= 1) {
+                //Ignores the string segment if the length is <= 1
+
+            } else if (nameKeywords[i].substring(0, 2).equals(TASK_DATE)) {
                 datePrefixLocation = i;
             }
```

Comment:
to shorten the code for this part, perhaps you can consider just checking for the second condition

Reply:
yes I understand where you are coming from but the first if condition is used to solve bug #256